frame/system/src/lib.rs

```diff
@@ -599,6 +599,11 @@ decl_module! {
 		fn set_heap_pages(origin, pages: u64) {
 			ensure_root(origin)?;
 			storage::unhashed::put_raw(well_known_keys::HEAP_PAGES, &pages.encode());
+			// is this unwrap legal?
+			let code: Vec<u8> = storage::unhashed::get(well_known_keys::CODE).unwrap();
```

Comment:
answer is "no" - `unwrap`s aren't legal anywhere in the codebase except for tests

Reply:
sure I put there as a reminder to figure out the more appropriate error to return in this scenario. Also out of curiosity under what conditions do you think `:code` isn't available in storage?

Reply:
Code will always be in `storage`, but `unwrap` is not allowed. Use `expect` instead.

Reply:
As an aside, this increases the execution weight significantly.

Reply:
> As an aside, this increases the execution weight significantly.

how significant?

Reply:
so we can understand if this approach is economically feasible

Reply:
I judge on the similarity of this to the `set_code` which takes up the whole block worth of weight. 

I am not sure exactly why it is that way, but I think it is because the contents of `:code` is generally considered to be rather large. Then, they are hashed as part of storage root calculation. (Then we store it in the db but I think this cost is not reflected in weight). Here it is similar because we also do the hashing and unlike the storing case, we load which does have some DB latency involved. 

IDK, maybe @shawntabrizi can say something about this here? In any case, the weight annotation and comments should be updated.

Reply:
paging dr. @shawntabrizi :grin:

Reply:
We need to benchmark this. How is this function used?

You can ping me on DM and I can help you with benchmarking.